src/ngff_tables_prototype/reader.py

```diff
@@ -9,52 +12,396 @@
+                np.ones((len(anndata_obj), 1), dtype=float),
+            ),
+            axis=1,
+        )
+        kwargs["face_color"] = random_colors
 
+    new_layer_data = (np.fliplr(xy), kwargs, "points")
+    return new_layer_data
+
+
+def get_napari_table_layer_data(file_path: str, table_group_key: str) -> LayerDataTuple:
+    ome_zarr = parse_url(os.path.join(file_path, table_group_key))
     # load table
     anndata_obj = load_table_to_anndata(file_path, table_group_key)
-    table_group = zarr.group(ome_zarr.store)[table_group_key]
-    if "label" in table_group.attrs:
-        features_table = _anndata_to_napari_labels_features(anndata_obj)
 
-        # get the label item
-        label_name = table_group.attrs["label"]
-        for i, (data, kwargs, layer_type) in enumerate(layer_data):
+    # then load the regions object associated to the table, necessary to map features to space
+    table_group = zarr.group(ome_zarr.store)
+    tables_region = table_group.attrs["region"]
+    if tables_region is None:
+        raise ValueError(
+            "every feature table must be associated with a regions object, otherwise there is no way to map the features to space"
+        )
+    tables_region_key = table_group.attrs["region_key"]
+    if tables_region_key is not None:
+        raise NotImplementedError(
+            'currently only a single "regions" object can be mapped to a feature table'
+        )
+    table_instance_key = table_group.attrs["instance_key"]
+    group_type = tables_region.split("/")[0]
+    if group_type == "labels":
+        features_table = _anndata_to_napari_labels_features(
+            anndata_obj, table_instance_key=table_instance_key
+        )
+        return get_napari_label_layer_data(
+            file_path=file_path,
+            labels_group_key=tables_region,
+            features_table=features_table,
+            tables_instance_key=table_instance_key,
+        )
+    elif group_type == "circles":
+        return get_napari_circles_layer_data(
+            file_path=file_path,
+            circles_group_key=tables_region,
+            features_table=anndata_obj.to_df(),
+            tables_instance_key=table_instance_key,
+        )
+    elif group_type == "polygons":
+        raise NotImplementedError()
+    else:
+        raise ValueError(
+            'region must specify the zarr path (relative to the root) of a "region" type'
+        )
+
+
+def get_napari_points_layer_data(
+    file_path: str, points_group_key: str
+) -> LayerDataTuple:
+    # load table
+    anndata_obj = load_table_to_anndata(file_path, points_group_key)
+
+    ome_zarr = parse_url(file_path)
+    table_group = zarr.group(ome_zarr.store)[points_group_key]
+    assert (
+        "@type" in table_group.attrs
+        and table_group.attrs["@type"] == "ngff:points_table"
+    )
+
+    # what is layer_properties for? notice that not all the layers set this
+    layer_properties = anndata_obj.obs
```

Comment:
layer properties is a table associated with the layer. for example, with the points layer, the table annotates each point. it can be used for things such as coloring the points (i.e., map point color to property value).